File: year2020/month10/day1007/ms_1007.cpp
#include <iostream>
#include <cmath>

using namespace std;

int ESP(int n); // Each Sum of Power

int main() {
	int n;

	cin >> n;

	if (0 < n && n < 1000000000)
		cout << ESP(n) << endl;

	return 0;
}

int ESP(int n) {
	if (n == 0)
		return 0;
	else
		return pow(n % 10, 2) + ESP(n / 10);
}

File: year2020/month10/day1007/ms_1007.cs
using System;

namespace algorithm_73
{
    class Program
    {
        static int ESP(int n) // Each Sum of Power
        {
            if(n == 0)
                return 0;
            else
                return Convert.ToInt32(Math.Pow(n % 10, 2)) + ESP(n/10);
        }

        static void Main(string[] args)
        {
            int n = Convert.ToInt32(Console.ReadLine());

            if(0 < n && n < 1000000000)
                Console.WriteLine(ESP(n));
        }
    }
}
